homeassistant/components/binary_sensor/ihc.py

```diff
@@ -0,0 +1,183 @@
+    """Auto setup ihc binary sensors from ihc project."""
+    _LOGGER.info("Auto setup - IHC Binary sensors")
+
+    def setup_product(ihcid, name, product, productcfg):
+        """Product setup callback."""
+        add_sensor_from_node(devices, ihcplatform.ihc, ihcid, name,
+                             product, productcfg['type'],
+                             productcfg['inverting'])
+    ihcplatform.autosetup(PRODUCTAUTOSETUP, setup_product)
+
+
+class IHCBinarySensor(IHCDevice, BinarySensorDevice):
+    """IHC Binary Sensor."""
+
+    def __init__(self, ihccontroller, name, ihcid, sensortype: str,
+                 inverting: bool, ihcname: str, ihcnote: str,
+                 ihcposition: str):
+        """Initialize the IHC binary sensor."""
+        IHCDevice.__init__(self, ihccontroller, name, ihcid, ihcname,
+                           ihcnote, ihcposition)
+        self._state = STATE_UNKNOWN
+        self._sensor_type = sensortype
+        self.inverting = inverting
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return self._sensor_type
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on/open."""
+        return self._state
+
+    def update(self):
+        """We do not nothing."""
+        pass
+
+    def on_ihc_change(self, ihcid, value):
+        """Callback when ihc resource changes."""
+        if self.inverting:
+            self._state = not value
+        else:
+            self._state = value
+        self.schedule_update_ha_state()
+
+
+def add_sensor_from_node(devices, ihccontroller, ihcid: int, name: str,
+                         product, sensortype,
+                         inverting: bool) -> IHCBinarySensor:
+    """Add a sensor from the ihc project node."""
+    ihcname = product.attrib['name']
+    ihcnote = product.attrib['note']
+    ihcposition = product.attrib['position']
+    return add_sensor(devices, ihccontroller, ihcid, name, sensortype, False,
+                      inverting, ihcname, ihcnote, ihcposition)
+
+
+def add_sensor(devices, ihccontroller, ihcid: int, name: str,
+               sensortype: str=None, overwrite: bool=False,
+               inverting: bool=False, ihcname: str="",
+               ihcnote: str="", ihcposition: str="") -> IHCBinarySensor:
+    """Add a new a sensor."""
+    if ihcid in _IHCBINARYSENSORS:
+        sensor = _IHCBINARYSENSORS[ihcid]
+        if overwrite:
+            sensor.set_name(name)
+            _LOGGER.info("IHC sensor set name: " + name + " " + str(ihcid))
+    else:
+        sensor = IHCBinarySensor(ihccontroller, name, ihcid, sensortype,
+                                 inverting, ihcname, ihcnote, ihcposition)
+        _IHCBINARYSENSORS[ihcid] = sensor
+        devices.append(sensor)
+        _LOGGER.info("IHC sensor added: " + name + " " + str(ihcid))
```

Comment:
See above, how to use string formatting in the logger.

Reply:
Done